tests/components/blueprint/test_websocket_api.py

```diff
@@ -80,3 +97,143 @@ async def test_import_blueprint(hass, aioclient_mock, hass_ws_client):
             },
         },
     }
+
+
+async def test_save_blueprint(hass, aioclient_mock, hass_ws_client):
+    """Test saving blueprints."""
+    raw_data = Path(
+        hass.config.path("blueprints/automation/test_event_service.yaml")
+    ).read_text()
+
+    with patch("pathlib.Path.write_text", return_value=Mock()) as write_mock:
+        client = await hass_ws_client(hass)
+        await client.send_json(
+            {
+                "id": 6,
+                "type": "blueprint/save",
+                "path": "test_save",
+                "data": raw_data,
+                "domain": "automation",
+                "source_url": "https://github.com/balloob/home-assistant-config/blob/main/blueprints/automation/motion_light.yaml",
+            }
+        )
+
+        msg = await client.receive_json()
+
+        assert msg["id"] == 6
+        assert msg["success"]
+        assert write_mock.mock_calls
+        assert write_mock.call_args[0] == (
+            "blueprint:\n  name: Call service based on event\n  domain: automation\n  input:\n    trigger_event:\n    service_to_call:\n  source_url: https://github.com/balloob/home-assistant-config/blob/main/blueprints/automation/motion_light.yaml\ntrigger:\n  platform: event\n  event_type: !placeholder 'trigger_event'\naction:\n  service: !placeholder 'service_to_call'\n",
```

Comment:
Lol. This will be a pain if we update `blueprints/automation/test_event_service.yaml` 😆

Reply:
(we probably don't)